src/main/java/de/medizininformatikinitiative/torch/util/Redaction.java

```diff
@@ -41,23 +41,15 @@ public Base redact(Base base) {
                 CanonicalType profileurl = resource.getMeta().getProfile().getFirst();
                 structureDefinition=CDS.getDefinition(resource.getMeta().getProfile());
                 if(structureDefinition==null){
-                    logger.warn("Unknown Profile {}",profileurl);
+                    logger.error("StructureDefinition is null for profile URL: {}", profileurl.getValue());
                     return new Patient();
+
                 }
-                if (!Objects.equals(profileurl.getValue(), structureDefinition.getUrl())) {
+                /*if (!Objects.equals(profileurl.getValue(), structureDefinition.getUrl())) {
                     logger.warn("Profile Missmatch {} {}", structureDefinition.getUrl(), profileurl.getValue());
-                }
-                // Check if structureDefinition is not null
-                if (structureDefinition != null) {
+                }*/
                     elementID = String.valueOf(resource.getResourceType());
```

Comment:
why is the resource type called elmentID ? - should it not be
typeElementID ?

Reply:
No it is the start of the recursive Redaction